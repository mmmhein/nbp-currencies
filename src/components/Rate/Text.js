import styled from 'styled-components';

import BaseText from '../Text';

const Text = styled(BaseText)`
  width: 100%;
`;

export default Text;
