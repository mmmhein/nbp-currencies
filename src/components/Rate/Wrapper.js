import styled from 'styled-components';

import { colorLightGrey, borderWidthThin, spacingMedium } from '../../styles/designTokens';

const Wrapper = styled.li`
  display: flex;
  align-items: center;
  border-bottom: ${borderWidthThin} solid ${colorLightGrey};
  padding: ${spacingMedium} 0;
`;

export default Wrapper;
