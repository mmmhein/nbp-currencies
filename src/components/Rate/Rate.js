import React from 'react';
import PropTypes from 'prop-types';

import { rate } from '../../types';
import Text from './Text';
import Wrapper from './Wrapper';

function Rate({ rate, children }) {
  const { currency, code, mid } = rate;

  return (
    <Wrapper data-testid="rate">
      <Text>{currency}</Text>
      <Text>{code}</Text>
      <Text bold>{mid}</Text>
      {children}
    </Wrapper>
  );
}

Rate.propTypes = {
  rate: rate.isRequired,
  children: PropTypes.node
};

export default Rate;
