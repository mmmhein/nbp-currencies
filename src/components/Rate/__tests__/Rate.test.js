import React from 'react';
import { create } from 'react-test-renderer';

import Rate from '../Rate';

jest.mock('../../Text', () => 'Text');

describe('COMPONENT - Rate', () => {
  it('renders correctly', () => {
    const rate = {
      currency: 'euro',
      code: 'euro',
      mid: 4
    };
    const component = create(<Rate rate={rate} />);

    expect(component.toJSON()).toMatchSnapshot();
  });
});
