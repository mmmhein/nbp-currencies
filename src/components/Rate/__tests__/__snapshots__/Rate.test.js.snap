// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`COMPONENT - Rate renders correctly 1`] = `
.c1 {
  width: 100%;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-bottom: 1px solid #b0b0b0;
  padding: 2rem 0;
}

<li
  className="c0"
  data-testid="rate"
>
  <Text
    className="c1"
  >
    euro
  </Text>
  <Text
    className="c1"
  >
    euro
  </Text>
  <Text
    bold={true}
    className="c1"
  >
    4
  </Text>
</li>
`;
